language/move-lang/functional-tests/tests/libra/wallet/wallet_module.move

```diff
@@ -0,0 +1,202 @@
+module ApprovalGroup {
+    use 0x1::Signature;
+
+    resource struct ApprovalGroup {
+        // we do not have collection support in Move now, so illustrate
+        // using 2 out-of 3
+
+        // for simplicity just use the plain public key here. We could
+        // also use the hash here as how auth key works
+        pk1: vector<u8>,
+        pk2: vector<u8>,
+        pk3: vector<u8>,
+
+        // the threshold policy
+        threshold: u64,
+
+        // Recipient address allowlist policy ...
+    }
+
+    // create a new approval group
+    public fun create(
+        pk1: vector<u8>,
+        pk2: vector<u8>,
+        pk3: vector<u8>,
+    ): Self::ApprovalGroup {
+        ApprovalGroup {pk1, pk2, pk3, threshold: 2}
+    }
+
+    // helper function to evaluate the threshold policy
+    fun verify_sig(
+        group: &Self::ApprovalGroup,
+        pk: vector<u8>,
+        sig: vector<u8>,
+        hash: vector<u8>,
+    ): bool {
+        if ((copy pk == *&group.pk1) ||
+            (copy pk == *&group.pk2) ||
+            (copy pk == *&group.pk3)) {
```

Comment:
small nit, do we need parens here around each eq?

Reply:
grammarly we don't, so let me remove them.